packages/dds/tree/src/simple-tree/core/allowedTypes.ts

```diff
@@ -131,7 +131,30 @@ export interface AllowedTypeMetadata {
 	 */
 	readonly custom?: unknown;
 
-	// TODO metadata for enablable types will be added here
+	/**
+	 * If defined, indicates that an allowed type is staged. Before upgrade, any attempt to insert or move a node to the location where this allowed type is declared will throw an error.
+	 */
+	readonly stagedSchemaUpgrade?: SchemaUpgrade;
+}
+
+/**
+ * Package internal construction API.
+ */
+export let createSchemaUpgrade: () => SchemaUpgrade;
+
+/**
+ * Unique token used to upgrade schemas and determine if a particular upgrade has been completed.
+ *
+ * TODO:#38722 implement runtime schema upgrades until then, the class purely behaves as a placeholder and we disable no-extraneous-class
+ * @sealed @alpha
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class SchemaUpgrade {
+	static {
+		createSchemaUpgrade = () => new SchemaUpgrade();
+	}
+
+	private constructor() {}
```

Comment:
The class is marked as `@sealed` - do we need to actually enforce that it can't be subclassed? It's confusing because we have many sealed classes elsewhere where we don't use this pattern. It's not standard in our package.